### PR TITLE
ECM-661 Add validation to the Reinstate Case function

### PR DESCRIPTION
**Pull request by cindychmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-661

### Change description ###

Add validation to the Reinstate Case action to ensure the user changes the current position from 'Case closed'.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseCloseServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Test
+    void shouldValidateReinstateClosedCase_IsCaseClose_ReturnOne() {
+        String errorMessage = "This case cannot be reinstated with a "
+                + "current position of Case closed. Please select a different current position.";
+        CaseData caseData = new CaseData();
+        caseData.setPositionType(CASE_CLOSED_POSITION);
+        List<String> errors = caseCloseService.validateReinstateClosedCaseMidEvent(caseData);
+        Assertions.assertEquals(1, errors.size());
```
Comment (hsjhita) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseCloseServiceTest.java:28]: nothing too big but maybe just import assertEquals

Comment (hsjhita) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseCloseServiceTest.java:28]: and case data could be moved to the setup

**Review by hsjhita — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseCloseServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @BeforeEach
+    public void setup() throws Exception {
+        caseCloseService = new CaseCloseService();
+        caseData = new CaseData();
+    }
+
+    @Test
+    void shouldValidateReinstateClosedCase_IsCaseClose_ReturnOne() {
+        String errorMessage = "This case cannot be reinstated with a "
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseCloseServiceTest.java:25]: A better approach is to make use of the constant you've defined in the service. If you remove the private modifier here:
`private static final String REINSTATE_CANNOT_CASE_CLOSED_ERROR_MESSAGE`

then you can use the variable in this unit test

`assertEquals(REINSTATE_CANNOT_CASE_CLOSED_ERROR_MESSAGE, errors.get(0));`

Comment (cindychmcts) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseCloseServiceTest.java]: Thank you James. Updated.

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java [hunk 3]
```diff
@@ -771,6 +776,35 @@ public void dynamicRestrictedReportingErrors() throws Exception {
...
+                .header("Authorization", AUTH_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.data", notNullValue()))
+                .andExpect(jsonPath("$.errors", notNullValue()))
+                .andExpect(jsonPath("$.warnings", nullValue()));
+    }
+
+    @Test
+    public void reinstateClosedCaseMidEventValidationValidationErrors() throws Exception {
```
Comment (btensay) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java:794]: How about this test reinstateClosedCaseMidEventValidationErrors

**Review by btensay — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java (hunk 3 above)
Comment (cindychmcts) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java]: Thank you Tensay. Updated.

**Pull request merged**